plugins/meta/tuning/tuning.go

```diff
@@ -165,6 +183,122 @@ func changeMtu(ifName string, mtu int) error {
 	return netlink.LinkSetMTU(link, mtu)
 }
 
+func createBackup(ifName, containerID, backupPath string, tuned bool, tuningConf *TuningConf) error {
+	config := configToRestore{Tuned: tuned}
+	if tuned {
+		link, err := netlink.LinkByName(ifName)
+		if err != nil {
+			return fmt.Errorf("failed to get %q: %v", ifName, err)
+		}
+		if tuningConf.Mac != "" {
+			config.Mac = link.Attrs().HardwareAddr.String()
+		}
+		if tuningConf.Promisc {
+			config.Promisc = new(bool)
+			*config.Promisc = (link.Attrs().Promisc != 0)
+		}
+		if tuningConf.Mtu != 0 {
+			config.Mtu = link.Attrs().MTU
+		}
+	}
+
+	backupDir := path.Join(backupPath, containerID)
+
+	if _, err := os.Stat(backupDir); os.IsNotExist(err) {
+		if err = os.MkdirAll(backupDir, 0600); err != nil {
+			return fmt.Errorf("failed to create backup directory: %v", err)
+		}
+	}
+
+	data, err := json.MarshalIndent(config, "", " ")
+	if err != nil {
+		return fmt.Errorf("failed to marshall data for %q: %v", ifName, err)
+	}
+	if err = ioutil.WriteFile(path.Join(backupDir, ifName+".json"), data, 0600); err != nil {
```

Comment:
Is it a wise choice to cache something inside container? Or `/var/lib/cni/` on host is a better place? which is just being used by IPAM plugin `host-local`.

Reply:
Backup file is stored in host's filesystem (`/var/lib/cni/tunig` by default). I think that storing backup in the container wouldn't be wise as container is not exactly persistent and might be for example deleted outside of the K8s. Additionally I think that if NIC is host's resource, therefore NIC's config should be stored on host itself.